src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -525,7 +525,7 @@ private boolean matchesComment(Pattern pattern, int lineNo) {
         final Matcher matcher = pattern.matcher(line);
         return matcher.find()
                 && getFileContents().hasIntersectionWithComment(
-                        lineNo, matcher.start(), lineNo, matcher.end());
+                        lineNo, -1, lineNo, matcher.end());
```

Comment:
@coderdeadpool please explain why we cannot kill this mutation with some code like this:
```
case 2:
     int fallthru = 2; // some comment
case 3: // should be violation
...

```

Reply:
Just do it

Reply:
@coderdeadpool, please prove that suggest input java code  does not kill mutation.

Reply:
You need to remove this code change to prove that suggested input is going to kill mutation

Reply:
@romani After adding the hardcoded mutation and the suggested test case. the Mutation still survived.
Does this means the test case does not able to kill the mutation ?

Reply:
Do only test extension, no change in main code.

Reply:
@romani getting this error, Dont kow why
```
patch -p1 << EOF
diff --git a/config/pitest-suppressions/pitest-coding-2-suppressions.xml b/config/pitest-suppressions/pitest-coding-2-suppressions.xml
index 7f7a16a..e4210a0 100644
--- a/config/pitest-suppressions/pitest-coding-2-suppressions.xml
+++ b/config/pitest-suppressions/pitest-coding-2-suppressions.xml
@@ -9,6 +9,[15](https://github.com/checkstyle/checkstyle/actions/runs/4612076083/jobs/8152562330?pr=12950#step:7:16) @@
     <lineContent>catchStmt = catchStmt.getNextSibling();</lineContent>
   </mutation>
 
+  <mutation unstable="false">
+    <sourceFile>FallThroughCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.coding.FallThroughCheck</mutatedClass>
+    <mutatedMethod>matchesComment</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator</mutator>
+    <description>removed call to java/util/regex/Matcher::start</description>
+    <lineContent>lineNo, matcher.start(), lineNo, matcher.end());</lineContent>
+  </mutation>
+
   <mutation unstable="false">
     <sourceFile>FinalLocalVariableCheck.java</sourceFile>
     <mutatedClass>com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck</mutatedClass>
EOF
Error: Process completed with exit code 1.
```
I was getting the same error in #12944 .

Reply:
Commit https://github.com/checkstyle/checkstyle/pull/12950/commits/3b72ce379907a5ad43d58c4ec1656415ded5d686

Results https://github.com/checkstyle/checkstyle/actions/runs/4612076083/jobs/8152562330?pr=12950

It doesn't kill mutation 

@nrmancuso , any other ideas how to kill it by test?

Reply:
Have we tried similar case, like:
```
  case 2:
       int fallthrough = 2; // fall thru
  case 3: // should be ok
      break;

```

My thinking is that since we are checking if regexp match has intersection with comment, some variation of matching regexp and matching/not matching comment should kill this mutation.

Reply:
@nrmancuso I tried that too. Still no good

Reply:
@romani @nrmancuso I tried so many variations in the comment but still no test fails in local.  it's possible that the mutation does not significantly impact the code's behavior.